tests/dht/test_providers.nim

```diff
@@ -35,34 +39,31 @@ proc initProvidersNode(
   let d = initDiscoveryNode(rng, privKey, address, bootstrapRecords)
   newProvidersProtocol(d)
 
-proc toPeerRecord(p: ProvidersProtocol) : PeerRecord =
-  ## hadle conversion between the two worlds
-
-  #NodeId is a keccak-256 hash created by keccak256.digest and stored as UInt256
-  let discNodeId = p.discovery.localNode.id
-  ## get it back to MDigest form
-  var digest: MDigest[256]
-  digest.data = discNodeId.toBytesBE
-  ## get into a MultiHash
-  var mh = MultiHash.init(multiCodec("keccak-256"), digest).orError(HashError).get()
-  result = PeerRecord.init(
-    peerId = PeerId.init(mh.data.buffer).get,
```

Comment:
Here, we need the `PeerId` in the `SignedPeerRecord` to match the PeerId derived from `SignedPeerRecord.envelope`. This means we need to remove the derivation of the `PeerId` from the discovery `NodeId`.

Reply:
Depends on how you interpret deterministic, it can be serialized to different formats (v0, v1; base32, base64, etc...). In fact our defaults are wrong as per the spec (last I checked) - here is the corresponding issue https://github.com/status-im/nim-libp2p/issues/53

Reply:
> Not sure if that helps, but to get the public key from the private one:
> `key.getPublicKey().tryGet()`

Please correct me if I'm wrong, but I believe this is what is already being done, albeit without the `tryGet`, which shouldn't be mandatory as it is a test.